app/scripts/metamask-controller.js

```diff
@@ -4646,6 +4650,26 @@ export default class MetamaskController extends EventEmitter {
     return keyring.isUnlocked();
   }
 
+  /**
+   * get preferred hardware wallet device name.
+   *
+   * @param deviceName - HardwareDeviceNames
+   * @param hdPath - string
+   * @returns {Promise<string>}
+   */
+  async getHardwareDeviceName(deviceName, hdPath) {
+    if (deviceName === HardwareDeviceNames.trezor) {
+      const keyring = await this.getKeyringForDevice(deviceName, hdPath);
+      const { minorVersion } = keyring.bridge;
+      // Specific case for OneKey devices, see `ONE_KEY_VIA_TREZOR_MINOR_VERSION` for further details.
+      if (minorVersion && minorVersion === ONE_KEY_VIA_TREZOR_MINOR_VERSION) {
+        return HardwareDeviceNames.oneKeyViaTrezor;
+      }
+    }
+
+    return deviceName;
+  }
```

Comment:
hmm, this function could lead to issues if used to resolve a device name for functions like [`getKeyringForDevice`](https://github.com/MetaMask/metamask-extension/blob/90873fd37ee36ccd03085aba27ff47c39a023b77/app/scripts/metamask-controller.js#L4559), since the "device" is more like a readable name we use to associate a specific Keyring type - while in this case it is being used more as a specific model (but that needs to resolve to the Trezor Keyring anyway)

Reply:
sorry, i dont understand what you mean, currently the function `getHardwareDevicename` only have been used in `connect-hardware/index.js` onUnlockAccounts method, and the purpose of it is used to log `OneKey via Trezor` metric into mixpanel when user import a OneKey via trezor USB.  and the function hasn't been used in `getKeyringForDevice` function. 

Maybe the function name of this method mislead, maybe i should change the functions name to `getDeviceNameForMetricLog` (and name may be too long), How do you think?

Reply:
I'm aware that it is not currently used in `getKeyringForDevice`, my concern was more related to a potential future misuse of the method. 

I think that changing the name to one that is more clearly related to metrics is a good idea!

Reply:
Ok, i will change the method name now.

Reply:
Has rename the function to `getDeviceNameForMetric`